examples/flutter_gallery/lib/gallery/updates.dart

```diff
@@ -0,0 +1,63 @@
+// Copyright 2016 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:async';
+
+import 'package:flutter/material.dart';
+import 'package:flutter/services.dart';
+
+abstract class UpdateService {
+  // Override this with your custom update service implementation
+  Future<String> fetchUpdateUrl();
+}
+
+class Updater extends StatefulWidget {
+  Updater({this.updateService, this.child, Key key}) : super(key: key);
+
+  final UpdateService updateService;
+  final Widget child;
+
+  @override
+  State createState() => new UpdaterState();
+}
+
+class UpdaterState extends State<Updater> {
+  @override
+  void initState() {
+    super.initState();
+    _checkForUpdates();
+  }
+
+  Future<Null> _checkForUpdates() async {
+    String updateUrl = await config.updateService.fetchUpdateUrl();
+    if (updateUrl != null) {
+      bool wantsUpdate = await showDialog(context: context, child: _buildDialog());
+      if (wantsUpdate) UrlLauncher.launch(updateUrl);
```

Comment:
if statement bodies should be on a line on their own, not on the same line as the expression